job-service-db/src/main/resources/procedures/getJobs.sql

```diff
@@ -183,6 +184,36 @@ BEGIN
          ELSE quote_ident(in_sort_field)
        END ||
         ' ' || CASE WHEN in_sort_ascending THEN 'ASC' ELSE 'DESC' END;
-    RETURN QUERY EXECUTE sql;
+
+    -- Create temporary table as a base to update the job progress
+    EXECUTE 'CREATE TEMPORARY TABLE get_job_temp ON COMMIT DROP AS ' || sql ;
+
+    -- Create a duplicate of that temporary table to store the results after the update
+
+    CREATE TEMPORARY TABLE new_table ON COMMIT DROP AS TABLE get_job_temp;
+
+
+
+    FOR jobId IN SELECT * FROM get_job_temp LOOP
+
+            -- Take out an exclusive update lock on the job row
+            PERFORM NULL FROM job j
+            WHERE j.partition_id = in_partition_id
+              AND j.job_id = jobId
+                FOR UPDATE;
+
+            -- Process outstanding job updates
+            PERFORM internal_update_job_progress(in_partition_id, jobId);
+            UPDATE new_table nt SET
+                    status = j.status,
+                    percentage_complete = j.percentage_complete,
+                    failure_details = j.failure_details
+            FROM job j WHERE nt.job_id = j.job_id;
+
+    END LOOP;
+
+
+    -- Execute and return the table created
+    RETURN QUERY EXECUTE 'select * from new_table';
```

Comment:
- Does dynamic SQL really need to used here?  It's not obvious to me why it would need to be used.
- PostgreSQL is technically free to return these rows in any order since one has not been specified.

Reply:
Point 1 complete. Point 2 still to be taken care of.